include/openssl/bio.h.in

```diff
@@ -713,7 +714,9 @@ int BIO_nwrite0(BIO *bio, char **buf);
 int BIO_nwrite(BIO *bio, char **buf, int num);
 
 const BIO_METHOD *BIO_s_mem(void);
+# ifndef OPENSSL_NO_DGRAM
```

Comment:
Currently the dgram support depends on BIO_ADDR which depends on sock. If we want to change this, it would require larger rework - especially - how to make BIO_ADDR not depend on sock support is quite unclear to me.

Anyway this is IMO subject for subsequent PRs if we want to do it.